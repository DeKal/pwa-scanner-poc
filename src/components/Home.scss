.home-container {
  text-align: center;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
  align-self: center;
  height: 100vh;

  .logo {
    height: 25vmin;
    pointer-events: none;
  }
}