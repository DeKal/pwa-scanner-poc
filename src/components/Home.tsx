import React, { ReactElement } from 'react'
import PropTypes from 'prop-types'
import 'components/Home.scss'
import Button from 'react-bootstrap/Button'
import { HomeProps } from 'types/props'
import logo from 'assets/logo.jpg'

const Home = ({ history, removeUserSession }: HomeProps): ReactElement => {
  return (
    <div className="home-container">
      <div>
        <img src={logo} className="logo mt-5 mb-5" alt="logo" />

        <div className="mt-3 mb-3">
          <h5 className="mb-2">Welcome to</h5>
          <h1 className="font-weight-bold text-dark">Foodsavers</h1>
        </div>
      </div>
      <div className="mb-4">
        <Button
          data-test-id="btn-new-session"
          className="mt-5 mb-5"
          variant="outline-dark"
          size="lg"
          block
          onClick={() => {
            removeUserSession()
            history.push('/user')
          }}
        >
          New session
        </Button>
      </div>
    </div>
  )
}

export default Home
Home.defaultProps = {
  history: {
    push: () => {},
  },
  removeUserSession: () => {},
}
Home.propTypes = {
  history: PropTypes.object,
  removeUserSession: PropTypes.func,
}
