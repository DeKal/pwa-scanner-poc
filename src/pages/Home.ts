import { connect } from 'react-redux'
import { withRouter } from 'react-router'

import Home from 'components/Home'
import { removeUserSession } from 'state/actions/home.action'

const mapDispatchToProps = (dispatch) => ({
  removeUserSession: () => dispatch(removeUserSession()),
})

export default connect(null, mapDispatchToProps)(withRouter(Home))
