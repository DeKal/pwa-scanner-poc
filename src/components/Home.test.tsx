import React from 'react'
import { render } from '@testing-library/react'
import Home from './Home'

test('renders a Home component correctly', () => {
  const { container } = render(<Home />)
  expect(container).toMatchSnapshot()
})
