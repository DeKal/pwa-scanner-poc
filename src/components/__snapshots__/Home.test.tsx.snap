// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Home component correctly 1`] = `
<div>
  <div
    class="home-container"
  >
    <div>
      <img
        alt="logo"
        class="logo margin-vertical-32"
        src="/images/logo.jpg"
      />
      <div
        class="margin-32"
      >
        <div
          class="f-normal margin-16"
        >
          Welcome to
        </div>
        <div
          class="f-big bold c-grey"
        >
          Foodsavers
        </div>
      </div>
    </div>
    <div>
      <button
        class="margin-vertical-32 btn btn-outline-dark btn-block btn-lg"
        data-test-id="btn-new-session"
        type="button"
      >
        New session
      </button>
    </div>
  </div>
</div>
`;
